mypyc/sametype.py

```diff
@@ -52,5 +52,11 @@ def visit_rtuple(self, left: RTuple) -> bool:
             and len(self.right.types) == len(left.types)
             and all(is_same_type(t1, t2) for t1, t2 in zip(left.types, self.right.types)))
 
+    def visit_rstruct(self, left: RStruct) -> bool:
+        return (isinstance(self.right, RStruct)
+            and self.right.names == left.names
```

Comment:
On second thought, I think the best we can do it require that the names of the structs are identical, since in C two different structs can't be compatible with each other.